### ISSUE TITLE
Disable right click tool popup in web browser when building with wasm-bindgen

### ISSUE THREAD

**Issue by Azkarell** [bevyengine/bevy#4721]
## What problem does this solve or what need does it fill?

It might not have the highest priority but i built a small minesweeper remake with bevy and deployed it with wasm-bindgen & github pages. Marking of fields is possible with right-click. but in browser this opens a tool window as well. 

## What solution would you like?

Supress the the right click popup in browser when building with wasm-bindgen.

## What alternative(s) have you considered?
My guess with some time and effort or a different build tool this can be fixed as well. But i really liked the simplicity of wasm-bindgen + wasm build and i think most if not all projects would benefit from this as well if they get deployed to web. For me it is kind of unexpected behaviour that a right click does something else as well.

## Additional context

[link to game](https://azkarell.github.io/rust_minesweeper/)

**Comment by Azkarell** [comment]
I am aware that it is possible to do this manually (and thanks for the link :D). But i do believe for a game engine this should be the "default" behaviour. This is at least my opinion from a users point of view. Everything i have to do make something "work" is less than optimal, especially if you have limited background with web technologies and aren't even aware that something might behave different. But i agree that you probably should focus on more pressing issues and only do this if it is easy. (that's why it was a feature request). Alternatively maybe there is way to provide an index.html that does this out of box that can then be updated. In my opinion it should be as easy as possible to support different platforms.

**Comment by bjorn3** [comment]
I personally have `dom.event.contextmenu.enabled` set to false in firefox, which forces the context menu to always work. I find disabling it annoying as it makes it harder to inspect a specific element.

> Marking of fields is possible with right-click.

Why would you use right-click for that rather than left-click? It is not like you are showing your own contextmenu. If I right-click I expect a contextmenu of some sort, not an arbitrary different action.

**Comment by Azkarell** [comment]
left click -> reveal the field; right click -> mark it as bomb or possible empty. currently not that helpful but if in the future i want to add different difficulties or dumb down the revealing algorithm that reveals currently a little bit too much it should be helpful. and as far as i remember the original minesweeper had this as well :).

**Comment by devil-ira** [comment]
How do other game-engines that let you publish to the web handle this?

@bjorn3 I always use Ctrl+Shift+C to inspect an element. Not sure if that works on Firefox.

**Comment by devil-ira** [comment]
Yeah, Bevy does not have tooling for publishing builds yet. There has been some talk about creating a cli for bevy to handle things like that.

**Comment by danec020** [comment]
Could I offer a suggestion of adding an overload/option on the WindowPlugin to disable context menu? This allows the developer to easily add this functionality to their game without having to worry about modifying other generated files. I also understand that this is part of web development and think any choice is fine but this one wouldn't be too bad IMO. This may be useful for other platforms that we aren't considering as well? Maybe not label it as a context menu fix but more of Window consume right click type of option. Documenting that it will send the right click to the game but on platforms where the platform tries to take control of this action, bevy will try to handle it.

If nothing is done in the engine for this, I think there should be an example that shows how to block the context menu on only the generated canvas that wasm-bindgen creates on init. I think blocking it for the entire page is a bad practice. Should never take over some ones ability to use their browser features unless you have an extremely necessary and good reason.

**Comment by daxpedda** [comment]
Winit does actually prevent the context menu by default by using [`Event.preventDefault()`](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault). I can see that Bevy [sets](https://github.com/bevyengine/bevy/blob/05b00267c60fb563ad5fbc64152b1108b24b2d34/crates/bevy_winit/src/winit_windows.rs#L143) it to [`true`](https://github.com/bevyengine/bevy/blob/05b00267c60fb563ad5fbc64152b1108b24b2d34/crates/bevy_window/src/window.rs#L270) by default.

As far as I can tell using the default options right now in Bevy, disables the context menu as well. So this issue can be closed?

Cc https://github.com/bevyengine/bevy/issues/4752.

**Comment by mockersf** [comment]
just tried on Safari and Firefox, the context menu is not blocked. Scrolling events are blocked, so the option is correctly enabled

**Comment by daxpedda** [comment]
> just tried on Safari and Firefox, the context menu is not blocked.

Can I ask what exactly you did? Because I can't seem to reproduce this on Linux nor on Windows for Firefox and Chrome.

I tried the `breakout` example and the following:
```rust
fn main() {
    App::new().add_plugins(DefaultPlugins).run()
}
```
just using the default cargo features. Both seem to correctly block the context menu.

**Comment by mockersf** [comment]
right clicking works for me on https://bevyengine.org/examples/Games/breakout/, built with Bevy 0.12.1, on chromium, safari and firefox, on macos

**Comment by daxpedda** [comment]
Yeah ... unfortunately my use of [`wasm-server-runner`](https://github.com/jakobhellermann/wasm-server-runner/blob/0c0e99714df5884a30c0b877ff4d192e0b7b1ed9/static/index.html#L175-L178) has blinded me.

See https://github.com/rust-windowing/winit/pull/3282. This will hopefully land in Winit v0.29.6.

**Comment by mockersf** [comment]
nice, thanks!

**Comment by janhohenheim** [comment]
Closing as this winit version has long since been adopted
